src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/mbscheduler_hwemu.cxx

```diff
@@ -1997,6 +2016,7 @@ namespace hwemu {
             scheduler_thread->join(); //! Wait untill scheduler_thread exits
         }
 
+        delete exec;

```

Comment:
This looks like a misplaced delete.  Please use std::unique_ptr for the resources .